### ISSUE TITLE
[Feature] specification for adapter is wanted

### ISSUE THREAD

**Issue by closetool** [casbin/casbin#827]
Casbin has many adapters. But their implementations has a lot of differences, like `ptype` and `p_type`, `casbin_rule` and `casbin_rules`. This is not good for development, we need to make a specification for it.

**Comment by closetool** [comment]
@Zxilly @nodece @hsluoyz

**Comment by Zxilly** [comment]
yes, also like filter has many implemetation. We must make a standard.

**Comment by hsluoyz** [comment]
@closetool we already had a conclusion for this:

1. https://github.com/casbin-rs/diesel-adapter/issues/17#issuecomment-770909065
2. https://github.com/casbin-rs/diesel-adapter/pull/58#issuecomment-769847842

Plz document this in our official site.
